spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/XDController.java

```diff
@@ -62,14 +64,20 @@
  * @author Gunnar Hillert
  */
 
-public abstract class XDController<D extends BaseDefinition, A extends ResourceAssemblerSupport<D, R>, R extends NamedResource> {
+public abstract class XDController<D extends BaseDefinition, A extends
+		ResourceAssemblerSupport<D, R>, R extends NamedResource, I extends BaseInstance<D>> {
 
-	private ResourceDeployer<D> deployer;
+	private AbstractInstancePersistingZKDeployer<D, I> deployer;
```

Comment:
If we want to defer this for now, we can add a comment as such:

```
	/*
	 * (non JavaDoc)
	 * TODO: This is declared as AbstractInstancePersistingZKDeployer
	 * because "deployer" is serving as both deployer and validator;
	 * these should be split into separate members even if implemented
	 * by the same object.
	 */
```